web/client-api/src/main/java/io/deephaven/web/client/fu/JsWeakRef.java

```diff
@@ -0,0 +1,16 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.web.client.fu;
+
+import jsinterop.annotations.JsPackage;
+import jsinterop.annotations.JsType;
+
+@JsType(namespace = JsPackage.GLOBAL, name = "WeakRef", isNative = true)
+public class JsWeakRef<T> {
+    public JsWeakRef(T target) {
+
+    }
```

Comment:
Does the constructor need to be defined here? I'm unsure how this works with the `isNative = true` flag.

Reply:
Yes, this is the native constructor and must be defined in java. Doing so lets Java know that this is the signature of the constructor, but since the type is marked `isNative=true`, this java impl will be ignored and the real JS call will be invoked instead.